packages/gatsby-transformer-sharp/src/customize-schema.js

```diff
@@ -374,6 +379,182 @@ const fluidNodeType = ({
   }
 }
 
+const imageNodeType = ({
+  pathPrefix,
+  getNodeAndSavePathDependency,
+  reporter,
+  cache,
+}) => {
+  return {
+    type: new GraphQLObjectType({
+      name: `ImageSharpGatsbyImage`,
+      fields: {
+        imageData: {
+          type: new GraphQLNonNull(GraphQLJSON),
+        },
+      },
+    }),
+    args: {
+      layout: {
+        type: ImageLayoutType,
+        defaultValue: `fixed`,
+        description: stripIndent`
+        The layout for the image.
+        FIXED: A static image sized, that does not resize according to the screen width
+        FLUID: The image resizes to fit its container. Pass a "sizes" option if it isn't going to be the full width of the screen. 
+        CONSTRAINED: Resizes to fit its container, up to a maximum width, at which point it will remain fixed in size.
+        `,
+      },
+      maxWidth: {
+        type: GraphQLInt,
+        description: stripIndent`
+        Maximum display width of generated files. 
+        The actual largest image resolution will be this value multipled by the largest value in outputPixelDensities
+        This only applies when layout = FLUID or CONSTRAINED. For other layout types, use "width"`,
+      },
+      maxHeight: {
+        type: GraphQLInt,
+      },
+      width: {
+        type: GraphQLInt,
+        description: stripIndent`
+        The display width of the generated image. 
+        The actual largest image resolution will be this value multipled by the largest value in outputPixelDensities
+        Ignored if layout = FLUID or CONSTRAINED, where you should use "maxWidth" instead.
+        `,
+      },
+      height: {
+        type: GraphQLInt,
+      },
```

Comment:
Do we warn if people are using it on the wrong layout?

Reply:
Yeah